src/redux/services/unsavedResource.ts

```diff
@@ -66,3 +66,51 @@ export function createUnsavedResource(
 
   return newResource;
 }
+
+export function createMultipleUnsavedResources(
+  inputs: {name: string; kind: string; apiVersion: string; namespace?: string; obj?: any}[],
+  dispatch: AppDispatch
+) {
+  const resourceMap: any = {};
+
+  inputs.forEach(input => {
+    const resourceID = uuidv4();
+    resourceMap[resourceID] = {};
+    resourceMap[resourceID].input = input;
+
+    if (input.obj) {
+      resourceMap[resourceID].content = {
+        ...input.obj,
+        apiVersion: input.apiVersion,
+        kind: input.kind,
+        metadata: {
+          ...(input.obj.metadata || {}),
+          name: input.name,
+          namespace: input.namespace,
+        },
+      };
+      resourceMap[resourceID].text = stringify(resourceMap[resourceID].content);
+    } else {
+      resourceMap[resourceID].text = createDefaultResourceText(input);
+      resourceMap[resourceID].content = parseYamlDocument(resourceMap[resourceID].text).toJS();
+    }
+  });
+
+  const newResources: K8sResource[] = Object.keys(resourceMap).map(resourceID => ({
```

Comment:
let's rename `resourceID` to `resourceId` to be consistent with other places where we use it